### ISSUE TITLE
Exit Conditions in initDeviceComms()

### ISSUE THREAD

**Issue by greggersaurus** [caseyokane/RaspberryProjects#3]
do you really want to exit the entire application if anything fails in this function? Maybe returning an error code would be better?

If you really do want to use exit(), call it with EXIT_FAILURE instead of "1"

**Comment by caseyokane** [comment]
Good point, I think when I did this I was under the impression that it would just exit the current function call rather than actually terminate the process (I really have no idea what I was thinking). Yeah It would probably be better to return an error code.

Would you suggest just including an extra parameter that is a pointer to an error code and then setting that value before returning (to terminate the current instantiation of the function). Or maybe just have the function return a value that can be categorized as an error (like the int -1 if I'm being overly simplistic).
